src/piadm.sh

```diff
@@ -512,14 +515,15 @@ install() {
 		# Explicit boot stamp or URL.
 
 		# Check if URL exists first
-		# XXX I think max-time 30s to fetch headers
+		# We believe max-time 30s to fetch headers
 		# is more than enough.
-		vecho "Checking if URL $1 exists"
+		vecho "Checking if URL $1 exists (30s) timeout"
 		if ! "${CURL[@]}" --max-time 30 --connect-timeout \
-				10 --head "$1" > /dev/null 2>&1; then
+				10 --head "$1" -w "%{http_code}" 2> /dev/null |\
+				tail -1 > /var/run/piadm.http.$$; then
 			# Get HTTP status code for error reporting
-			http_code=$(curl -ks -w "%{http_code}" --max-time 30 --connect-timeout \
-				10 --head -o /dev/null "$1" 2>/dev/null)
+			http_code=$(cat /var/run/piadm.$$)
+			/bin/rm -f /var/run/piadm.http.$$
```

Comment:
You leak this file if the check succeeds too.  Remove it in the `else` branch as well, please.